index.html

```diff
@@ -154,68 +168,107 @@ <h5>Warframe Drop Data</h5>
+            for (let i = 0; i < amount; i++) {
+                $("#tablebody").append(`<tr>
+                    <td>${data[i].item}</td>
+                    <td>${data[i].place}</td>
+                    <td>${data[i].rarity} (<b>${data[i].chance}%</b>)</td>
+                </tr>`)
+            }
 
-            console.log("./data/info.json", data)
+            $("#tablebody").append(`<tr><td class="msg" colspan="3"><strong>${data.length - amount}</strong> more results found. Refine your query to see more relevant results.</td></tr>`)
+        } else {
+            for (const obj of data) {
+                $("#tablebody").append(`<tr>
+                    <td>${obj.item}</td>
+                    <td>${obj.place}</td>
+                    <td>${obj.rarity} (<b>${obj.chance}%</b>)</td>
+                </tr>`)
+            }
+        }
 
-            let oldData = JSON.parse(localStorage.getItem("_wfinfo") || "{\"hash\":\"clem\"}")
+        $("th").on("click", function() {
+            if (this.id) { fill(data, this.id, sort === this.id ? !reverse : false, amount) }
+        })
+    }
 
-            if (hash !== oldData.hash) {
-                console.log("new hash found, re-download data")
-                localStorage.setItem("_wfinfo", JSON.stringify(data))
+    function search(searchValue) {
+        if (searchValue.length < 2) {
+            window.location.hash = ''
 
-                $.getJSON(`./data/all.slim.json?${time}`, function(all) {
-                    localStorage.setItem("_wfdata", JSON.stringify(all))
-                    onDataRetrieved(all)
-                })
-            } else {
-                onDataRetrieved(JSON.parse(localStorage.getItem("_wfdata")))
-            }
-        })
-    })
+            $("#tablebody").empty()
+            $("#tablebody").append("<tr><td class='msg'>You have to type in at least 2 characters to search.</td></tr>")
+            return
+        }
+
+        const searchArray = searchValue.split(' ').filter(item => item.length > 1)
+        const type = $('#search-type').val()
+        const amount = $('#display-amount').val()
+        let items = []
+
+        // Setup the search regex.
+        let regex = ''
+        for (const searchTerm of searchArray) {
+            regex += `(?=.*\\b.*${searchTerm}.*\\b)`
+        }
+        regex = RegExp(regex, 'gi')
+
+        window.location.hash = `/search/${encodeURIComponent(searchValue)}/${type}`
+
+        if (type === 'items')           { items = items.concat(window._data.filter(entry => regex.test(entry.item))) }
+        else if (type === 'locations')  { items = items.concat(window._data.filter(entry => regex.test(entry.place))) }
+        else if (type === 'both')       { items = items.concat(window._data.filter(entry => regex.test(entry.item) || regex.test(entry.place))) }
+        else                            { items = items.concat(window._data.filter(entry => regex.test(entry.item))) }
```

Comment:
please do actual blocks. this is fine for one statement, but more than one and they should actually be broken out and not on one line.

Reply:
This also looks like a lot of code duplication to me 🤔

Reply:
mostly like something we didn't catch when the item search options were implemented